src/app/forms/components/defects/defects.component.ts

```diff
@@ -51,20 +48,21 @@ export class DefectsComponent implements OnInit, OnDestroy {
     return this.defectsForm?.controls.length;
   }
 
-  trackByFn = (index: number): number => index;
-
-  getDefectForm = (i: number): CustomFormGroup => this.defectsForm?.controls[i] as CustomFormGroup;
-
-  getDefect(i: number): Defect | undefined {
-    const defectForm = this.getDefectForm(i);
-    const imNumber = defectForm.get(['imNumber'])?.value;
+  get testDefects(): TestResultDefect[] {
+    let defectArray: TestResultDefect[] = [];
 
-    return imNumber && this.defects?.find(defect => defect.imNumber === imNumber);
+    for (let i = 0; i < this.defectCount; i++) {
+      const data = this.defectsForm.controls[i] as CustomFormGroup;
+      const defect = data.getCleanValue(data) as TestResultDefect;
+      defectArray = [...defectArray, defect];
+    }
+    return defectArray;
   }
```

Comment:
This whole thing can be summarized into a `map()`.
```suggestion
  get testDefects(): TestResultDefect[] {
    return this.defectsForm.controls.map(control => {
      const formGroup = control as CustomFormGroup;
      return formGroup.getCleanValue(formGroup) as TestResultDefect;
    });
  }
```